src/storage.cr

```diff
@@ -428,12 +428,21 @@ class Storage
     end
   end
 
-  def mark_unavailable
+  # Limit mark targets with given arguments
+  # They should be checked again if they are really gone,
+  #   since they would be available which are renamed or moved
+  def mark_unavailable(ids_candidates : Array(String)?,
+                       titles_candidates : Array(String)?)
     MainFiber.run do
       get_db do |db|
         # Detect dangling entry IDs
         trash_ids = [] of String
-        db.query "select path, id from ids where unavailable = 0" do |rs|
+        # Use query builder instead?
```

Comment:
Yes ideally we should use query builder! I was just being lazy :P

Reply:
I understand :) I tried with [izniburak/query-builder](https://github.com/izniburak/query-builder), but, it's old, it has a syntax error.

Reply:
Oh I see. Let's worry about this later then.